### PR TITLE
More cleanup in the production logic

### PR DESCRIPTION
**Pull request by abcdefg30**
Depends on #10532 (the first five commits in here).

### CODE REVIEW

**Comment by abcdefg30** [comment]
Rebased as #10532 was merged.

**Comment by chrisforbes** [comment]
Generally :+1:

**Comment by abcdefg30** [comment]
@chrisforbes added your request in a new commit (can be squashed if wished).

File: OpenRA.Mods.Common/Traits/Player/ClassicProductionQueue.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class ClassicProductionQueueInfo : ProductionQueueInfo, Requires<TechTree
 		public override object Create(ActorInitializer init) { return new ClassicProductionQueue(init, this); }
 	}
 
-	public class ClassicProductionQueue : ProductionQueue, ISync
+	public class ClassicProductionQueue : ProductionQueue
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Player/ClassicProductionQueue.cs]: You need to get rid of https://github.com/OpenRA/OpenRA/pull/10557/files#diff-910e0462c780643261207e2829e4041dR47 as well then.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Player/ClassicProductionQueue.cs]: Nevermind, didn't read the commit message.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f5c1b12e6cbf8cf7eb768404c12a947257fef0d8...32615f5a56e4546db506a20f4b8c70412d5bace2)

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/26b97e59aa156f428fa41af8456c0c2c22a3f790...82a5d527786433f6129a341dc4e8c173f0277434)

**Pull request merged**